### ISSUE TITLE
[MPIR-248] NPE while DependenciesReport

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-project-info-reports-plugin#167]
**[Tony Chemit](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=tchemit)** opened **[MPIR-248](https://issues.apache.org/jira/browse/MPIR-248?redirect=false)** and commented

See trace

```
Caused by: java.lang.NullPointerException
	at org.apache.maven.report.projectinfo.dependencies.Dependencies.getJarDependencyDetails(Dependencies.java:243)
	at org.apache.maven.report.projectinfo.dependencies.renderer.DependenciesRenderer.hasSealed(DependenciesRenderer.java:1453)
	at org.apache.maven.report.projectinfo.dependencies.renderer.DependenciesRenderer.renderSectionDependencyFileDetails(DependenciesRenderer.java:547)
	at org.apache.maven.report.projectinfo.dependencies.renderer.DependenciesRenderer.renderBody(DependenciesRenderer.java:274)
	at org.apache.maven.reporting.AbstractMavenReportRenderer.render(AbstractMavenReportRenderer.java:79)
	at org.apache.maven.report.projectinfo.DependenciesReport.executeReport(DependenciesReport.java:186)

```

---

**Affects:** 2.5

**Issue Links:**
- [MPIR-275](https://issues.apache.org/jira/browse/MPIR-275) NPE in DependenciesReport

**Comment by jira-importer** [comment]
**[Tony Chemit](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=tchemit)** commented

With version 2.4, the problem does not occurs.

I will add a test for this, but I blame commit at first glance [commit 1103555](http://svn.apache.org/viewvc?rev=1103555&view=rev)

**Comment by jira-importer** [comment]
**[Tony Chemit](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=tchemit)** commented

System scope dependencies with no artifact file were not exclude.

[Fixed in rev 1370872](http://svn.apache.org/viewvc?rev=1370872&view=rev)

**Comment by jira-importer** [comment]
**[Ian Brandt](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ian@ianbrandt.com)** commented

I've hit this same stack trace with maven-project-info-reports-plugin 2.6.  See: MPIR-275.
